interp/interp.go

```diff
@@ -0,0 +1,166 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package interp
+
+import (
+	"fmt"
+	"math"
+	"sort"
+
+	"gonum.org/v1/gonum/spatial/r1"
+)
+
+// Interpolator1D represents a 1-dimensional interpolator.
+// It interpolates a function over a defined closed interval range
+// (which can be infinite on both sides).
+type Interpolator1D interface {
+	// Interval returns the valid closed interval for interpolation.
```

Comment:
If it's going away this ceases to be an issue.

Reply:
It makes a difference for piecewise-continuous interpolation. Also, `r1.Interval` has fields `Min` and `Max`, which to me suggests an interval closed on both sides. I would rather keep "closed", it seems to me the most common use case.

Reply:
We actually carefully avoided stating whether the `r1.Interval` types talks about closed or open end points.

Reply:
An open end point cannot be a maximum, since a maximum of a set is its largest element. An open interval doesn't have a largest element.

Am I being too picky about maths terminology here? ;-)

Reply:
We don't say whether the `r1.Interval{a, b}` is [a, b], (a, b), [a, b) or (a, b]. This is left up to the user.

Reply:
OK, I guess it's just me for whom the names `Min` and `Max` suggest a closed interval.